### PR TITLE
Auto add missing path parameters

### PR DESCRIPTION
**Pull request by puggan**
If there is path-parameters that are not define, automaticly add them as type string.

### CODE REVIEW

File: src/Attributes/Route.php [hunk 1]
```diff
@@ -73,6 +73,24 @@ function (array $params) {
...
+            }
+            if ($pathParms) {
+                foreach ($pathParms as $pathParm) {
+                    $param = new Parameter();
+                    $param->setName($pathParm);
+                    $param->setParamType('string');
+                    $this->getParams[] = $param;
+                }
+            }
+        }
```
Comment (uderline) [inline src/Attributes/Route.php:93]: I tried your code and it didn't work. This is my proposition:
```php
if (preg_match_all('#{([^}]+)}#', $this->route, $matches)) {
    $pathParams = $matches[1];
    $declaredParamsName = array_map(fn(Parameter $parameter): string => $parameter->getName(), ...$getParams);
    $unresolvedParams = array_diff($pathParams, $declaredParamsName);

    foreach ($unresolvedParams as $pathParam) {
        $param = new Parameter();
        $param->setName($pathParam);
        $param->setParamType(Type::STRING);
        $this->getParams[] = $param;
    }
}
```

What do you think ?

Comment (puggan) [reply src/Attributes/Route.php:93]: Ah, $getParams is a `Parameter[][]` not a `Parameter[]`, they are all documented as just array, so got no warning about it, and as my function had zero parameters, I just got an empty array anyway.

Comment (puggan) [reply src/Attributes/Route.php:93]: Your is a little better then mine, as it works for zero or one parameters, but if I remember how array_map works with multiple array parameters, it send them as separate parameters to the callback, so `$declaredParamsName` only contains the first parameters name.

Comment (puggan) [reply src/Attributes/Route.php]: Using your version, but with `$this->getParams` instead, seams to work.

https://github.com/uderline/openapi-php-attributes/blob/f1d45fda6083eb45e8dd7c3fa27ed7798877831f/src/Attributes/Route.php#L84

But alternative, we could flatten the params earlier, see PR #15 as an alternative.

I think I prefer #15 over this #12, but I let you decide how you want it.

**Comment by uderline** [comment]
Definitely go with #15

**Pull request closed without merging**